examples/deployments/aws-terraform/chroma.tf

```diff
@@ -102,26 +102,41 @@ resource "aws_instance" "chroma_instance" {
 }
 
 
-resource "aws_ebs_volume" "chroma-volume" {
+resource "aws_ebs_volume" "chroma_volume" {
+  count = var.prevent_chroma_data_volume_delete ? 0 : 1
+
   availability_zone = aws_instance.chroma_instance.availability_zone
   size              = var.chroma_data_volume_size
 
   tags = {
     Name = "chroma"
   }
+}
+
+resource "aws_ebs_volume" "chroma_volume_prevent_destroy" {
+  count = var.prevent_chroma_data_volume_delete ? 1 : 0
```

Comment:
Could you explain to me what this does? I couldn't find documentation on it and I'm wondering how it prevents deletion. I'm wondering if we should just remove the variable altogether and rely on users adjusting the lifecycle hook.

Reply:
The root is with TF as it has for some design decisions, made the choice not to allow evaluations within lifecycle hooks. So this is a very hacky way to work around that, but recently, I realized there is a better way to do this via snapshots on terminate. If the user wants guarantees that his data will not be lost if, for some reason, they run `destroy` without a backup/snapshot, aws provider has a nice `snapshot on terminate` mechanism (maybe it is a aws thing, actually). I've made that change and even removed the lifecycle hook altogether in this PR - https://github.com/chroma-core/chroma/pull/1173 

So I think this might be safe to close.